addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,381 @@
+        response_data = self._l10n_eg_eta_connect_to_server(request_data, request_url, 'GET', production_enviroment=invoice.company_id.l10n_eg_production_env)
+        if response_data.get('error'):
+            return response_data
+        response_data = response_data.get('response')
+        _logger.warning('PDF Function Response %s.', response_data)
+        if response_data.ok:
+            return {'data': response_data.content}
+        else:
+            return {'error': _('PDF Document is not available')}
+
+    @api.model
+    def _l10n_eg_validate_info_address(self, partner_id, issuer=False, invoice=False):
+        fields = ["country_id",
+                  "state_id", "city", "street",
+                  "l10n_eg_building_no"]
+        if (invoice and invoice.amount_total >= invoice.company_id.l10n_eg_invoicing_threshold) or self._l10n_eg_get_partner_tax_type(partner_id, issuer) != 'P':
+            fields.append('vat')
+        return all(partner_id[field] for field in fields)
+
+    @api.model
+    def _l10n_eg_get_eta_invoice_pdf(self, invoice):
+        return self._l10n_eg_get_eta_invoice_pdf_request(invoice)
+
+    @api.model
+    def _l10n_eg_eta_prepare_eta_invoice(self, invoice):
+        def group_tax_retention(tax_values):
+            return {'l10n_eg_eta_code': tax_values['tax_id'].l10n_eg_eta_code.split('_')[0]}
+
+        date_string = invoice.invoice_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+        grouped_taxes = invoice._prepare_edi_tax_details(grouping_key_generator=group_tax_retention)
+        invoice_line_data, totals = self._l10n_eg_eta_prepare_invoice_lines_data(invoice, grouped_taxes['invoice_line_tax_details'])
+        eta_invoice = {
+            'issuer': self._l10n_eg_eta_prepare_address_data(invoice.journal_id.l10n_eg_branch_id, invoice, issuer=True,),
+            'receiver': self._l10n_eg_eta_prepare_address_data(invoice.partner_id, invoice),
+            'documentType': 'i' if invoice.move_type == 'out_invoice' else 'c' if invoice.move_type == 'out_refund' else 'd' if invoice.move_type == 'in_refund' else '',
+            'documentTypeVersion': '1.0',
+            'dateTimeIssued': date_string,
+            'taxpayerActivityCode': invoice.journal_id.l10n_eg_activity_type_id.code,
+            'internalID': invoice.name,
+        }
+        eta_invoice.update({
+            'invoiceLines': invoice_line_data,
+            'taxTotals': [{
+                'taxType': tax['l10n_eg_eta_code'].split('_')[0].upper(),
+                'amount': round(abs(tax['tax_amount']), 5),
+            } for tax in grouped_taxes['tax_details'].values()],
+            'totalDiscountAmount': round(totals['discount_total'], 5),
+            'totalSalesAmount': round(totals['total_price_subtotal_before_discount'], 5),
+            'netAmount': round(totals['total_tax_exc'], 5),
+            'totalAmount': round(totals['total_tax_inc'], 5),
```

Comment:
should be the same as abs(invoice.amount_total_signed) and we also have amount_untaxed_signed...